sql/database.go

```diff
@@ -748,22 +844,34 @@ func (db *sqliteDatabase) copyMigrateDB(config *conf) (finalDB *sqliteDatabase,
 		WithConnections(1),
 		WithTemp(),
 		WithDatabaseSchema(config.schema),
+		WithExclusive(),
 	}
 	if !config.checkSchemaDrift {
 		opts = append(opts, WithNoCheckSchemaDrift())
 	}
 	migratedDB, err := Open("file:"+migratedPath, opts...)
 	if err != nil {
-		return nil, errors.Join(
-			fmt.Errorf("process temporary DB %s: %w", migratedPath, err),
-			deleteDB(migratedPath))
+		if err := deleteDB(migratedPath); err != nil {
+			config.logger.Error(
+				"incomplete temporary copy of the database couldn't be deleted",
+				zap.String("path", migratedPath),
+				zap.Error(err),
+			)
+		}
+		return nil, fmt.Errorf("process temporary DB %s: %w", migratedPath, err)
 	}
 	defer migratedDB.Close()
 	tempDBReady := false
 	defer func() {
-		err = errors.Join(err, migratedDB.Close())
+		migratedDB.Close()
 		if !tempDBReady {
-			err = errors.Join(err, deleteDB(migratedPath))
+			if err := deleteDB(migratedPath); err != nil {
+				config.logger.Error(
+					"incomplete temporary copy of the database couldn't be deleted",
+					zap.String("path", migratedPath),
+					zap.Error(err),
+				)
+			}
 		}
 	}()
```

Comment:
I think this can be simplified to

```go
defer migratedDB.Close()
```

and add

```go
		if err := deleteDB(migratedPath); err != nil {
			config.logger.Error(
				"incomplete temporary copy of the database couldn't be deleted",
				zap.String("path", migratedPath),
				zap.Error(err),
			)
		}
```

to the to blocks below where `tempDBReady` would be `false`. Wdyt?

Reply:
Fixed